src/NuGet.Core/NuGet.Packaging/Rules/DependenciesGroupsForEachTFMRule.cs

```diff
@@ -0,0 +1,206 @@
+                    noExactMatchString.Append(AnalysisResources.DependenciesGroupsForEachTFMEndingToNuspec);
+                    noExactMatchString.Append("\n");
+                    noExactMatchString.Append(AnalysisResources.DependenciesGroupsForEachTFMBeginningToNuspec);
+                    noExactMatchString.Append(" ");
+                    noExactMatchString.Append(tfm.GetFrameworkString());
+                }
+                noExactMatchString.Append(" ");
+                noExactMatchString.Append(AnalysisResources.DependenciesGroupsForEachTFMEndingToNuspec);
+            }
+
+            if (noExactMatchesFromNuspec.Count != 0)
+            {
+                if (noExactMatchString.Length == 0)
+                {
+                    noExactMatchString.Append(MessageFormat);
+                    noExactMatchString.Append("\n");
+                }
+                else
+                {
+                    noExactMatchString.Append("\n");
+                }
+
+                foreach (var tfm in noExactMatchesFromNuspec)
+                {
+                    if (tfm == noExactMatchesFromNuspec.First())
+                    {
+                        noExactMatchString.Append(AnalysisResources.DependenciesGroupsForEachTFMBeginningToFiles);
+                        noExactMatchString.Append(" ");
+                        noExactMatchString.Append(tfm.GetShortFolderName());
+                        continue;
+                    }
+                    noExactMatchString.Append(" ");
+                    noExactMatchString.Append(AnalysisResources.DependenciesGroupsForEachTFMEndingToFile);
+                    noExactMatchString.Append("\n");
+                    noExactMatchString.Append(AnalysisResources.DependenciesGroupsForEachTFMBeginningToFiles);
+                    noExactMatchString.Append(" ");
+                    noExactMatchString.Append(tfm.GetShortFolderName());
+                }
+                noExactMatchString.Append(" ");
+                noExactMatchString.Append(AnalysisResources.DependenciesGroupsForEachTFMEndingToFile);
+            }
+
+            if (compatNotExactMatches.Count != 0)
+            {
+                compatMatchString.Append(CompatMatchFoundWarningMessageFormat);
+                compatMatchString.Append("\n");
+                foreach (var tfm in compatNotExactMatches)
+                {
+                    if (tfm == compatNotExactMatches.First())
+                    {
+                        compatMatchString.Append(AnalysisResources.DependenciesGroupsForEachTFMBeginningToNuspec);
+                        compatMatchString.Append(" ");
+                        compatMatchString.Append(tfm.GetFrameworkString());
+                        continue;
+                    }
+                    compatMatchString.Append(" ");
+                    compatMatchString.Append(AnalysisResources.DependenciesGroupsForEachTFMEndingToNuspec);
+                    compatMatchString.Append("\n");
```

Comment:
I really think we should be using Environment.Newline everywhere. 
It's more exact.

Reply:
Exact how? As I wrote in the other comment, it's a TextWriter's responsibility (Console.Out is a TextWriter, but StreamWriter also implements it when writing to files) to write the correct new line ending. Over the weekend I verified it works as I expected (although I wasn't very careful, I trusted VSCode when it claimed the file used CRLF)

Reply:
Because you don't rely on other components behavior to handle it. 
Also string literals are ugly and error prone.

Reply:
I spent some time digging through TextWriter's docs and implementation, and surprise, surprise, I was wrong. There doesn't appear to be anything to convert `\n` to the environment new line. I'm really surprised because `dotnet run > out.txt` seems to use the Windows line ending, so I guess the shell's output redirection is putting it there.

Anyway, as Nikolche said, we should use `.AppendLine()` instead, and use `Environment.NewLine` in the test expected output.